### DISCUSSION TITLE
Oxide as WASM Module?

### DISCUSSION THREAD

**Discussion by bjesuiter** [tailwindlabs/tailwindcss#11610] [Ideas]
Hi there, 

could this oxide engine be compiled as wasm to make it usable in edge environments, like deno deploy? 

I Experiment heavily with the whole deno ecosystem and especially their web framework, deno fresh, and I like it  very much. 
One difference from node is the fully runtime/buildless workflow of most things in deno. 

They currently use twind as a runtime based tailwind clone in the "Fresh" framework, which is not optimal, bc. It misses certain tailwind features and needs to be kept up-to-date regularly with the upstream tailwind repo.

**Comment by adamwathan** [comment]
Yeah this is something I expect we will be digging into in the near future since we'll need it to properly deploy the oxide engine to play.tailwindcss.com 👍

**Reply by bjesuiter** [reply]
Nice to hear that! If you have anything like a PR or something to test, I'd be interested in beta testing :)

**Comment by jlukic** [comment]
Very curious on this. Was trying to reverse engineer how play.tailwindcss.com is generating class names, looks like a brotli and lightningcss wasm file are being used. 

Any chance you can share what your workaround has been so far?

Also curious on the package [@tailwindcss/oxide-wasm32-wasi](https://www.npmjs.com/package/@tailwindcss/oxide-wasm32-wasi) it appears to have a browser wasm, but haven't been able to get it to work (seeing some errors related to SharedArrayBuffer transfer from wasm related to threading).

**Comment by jlukic** [comment]
I've gone ahead and submitted a PR for browser wasm support https://github.com/tailwindlabs/tailwindcss/pull/18292
